### DISCUSSION TITLE
How to mock Request<Incoming>?

### DISCUSSION THREAD

**Discussion by gc-victor** [hyperium/hyper#3435] [Q&A]
I am upgrading to version 1, so I must modify my tests. I am having trouble mocking Request<Incoming>. Previously, it was easy to mock a Request and set a body. Please, could you give me any advice? Thanks!

**Comment by seanmonstar** [comment]
An `Incoming` is not meant to be constructed, similar to how `std::net::Incoming` isn't. Your best option would be to make your handler generic over the body type, then you can pass it custom bodies for tests.

```rust
fn hello<B>(req: Request<B>) -> Result<Response<MyRespBody>, MyError>
where
    B: Body,
{

}
```
